api/agentic/v1alpha1/accesspolicy_types.go

```diff
@@ -0,0 +1,179 @@
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+// AccessPolicySpec defines the desired state of AccessPolicy.
+type AccessPolicySpec struct {
+	// TargetRefs specifies the targets of the AccessPolicy.
+	// Currently, only Backend can be used as a target.
+	// +required
+	TargetRefs []gwapiv1.LocalPolicyTargetReference `json:"targetRefs"`
+	// Rules defines a list of rules to be applied to the target.
+	// +required
+	Rules []AccessRule `json:"rules"`
+}
+
+// AccessRule specifies an authorization rule for the targeted backend.
+// If the tool list is empty, the rule denies access to all tools from Source.
+type AccessRule struct {
+	// Source specifies the source of the request.
+	// +required
+	Source Source `json:"source"`
+	// Tools specifies a list of tools.
+	// +optional
+	Tools []string `json:"tools,omitempty"`
+}
+
+// Source specifies the source of a request.
+// This struct is same as the Source struct defined in https://github.com/kubernetes-sigs/gateway-api/blob/950c6639afd099b7bba4236f8b894ae4b891d26a/geps/gep-3779/index.md#api-design.
+//
+// At least one field may be set. If multiple fields are set,
+// a request matches this Source if it matches
+// **any** of the specified criteria (logical OR across fields).
+//
+// For example, if both `Identities` and `ServiceAccounts` are provided,
+// the rule matches a request if either:
+// - the request's identity is in `Identities`
+// - OR the request's Serviceaccount matches an entry in `ServiceAccounts`.
+//
+// Each list within the fields (e.g. `Identities`) is itself an OR list.
+//
+// If this struct is omitted in a rule, it matches any source.
+//
+// <gateway:util:excludeFromCRD> NOTE: In the future, if there’s a need to express more complex
+// logical conditions (e.g. requiring a request to match multiple
+// criteria simultaneously—logical AND), we may evolve this API
+// to support richer match expressions or logical operators. </gateway:util:excludeFromCRD>
+type Source struct {
+
+	// Identities specifies a list of identities that are matched by this rule.
+	// A request's identity must be present in this list to match the rule.
+	//
+	// Identities must be specified as SPIFFE-formatted URIs following the pattern:
+	//   spiffe://<trust_domain>/<workload-identifier>
+	//
+	// While the exact workload identifier structure is implementation-specific,
+	// implementations are encouraged to follow the convention of
+	// `spiffe://<trust_domain>/ns/<namespace>/sa/<serviceaccount>`
+	// when representing Kubernetes workload identities.
+	//
+	// +optional
+	Identities []string `json:"identities,omitempty"`
+
+	// ServiceAccounts specifies a list of Kubernetes Service Accounts that are
+	// matched by this rule. A request originating from a pod associated with
+	// one of these Serviceaccounts will match the rule.
+	//
+	// Values must be in one of the following formats:
+	//   - "<namespace>/<serviceaccount-name>": A specific Serviceaccount in a namespace.
```

Comment:
do we want to create a microsyntax or just make this a struct? Microsyntax is harder to change and evolve later